### ISSUE TITLE
No compatibility between node and bun for `new Error().stack` when I patched `prepareStackTrace`.

### ISSUE THREAD

**Issue by baseballyama** [oven-sh/bun#4606]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Run below code with `bun` command and `node` command.

```js
Error.prepareStackTrace = (_, stack) => stack;
const stack = new Error().stack.slice(1);
console.log(stack);
```

### What is the expected behavior?

Print `[ CallSite {} ]`.

### What do you see instead?

Print this.

```txt
rror: 
    at module code (/path/to/a.js:2:17)
```

### Additional information

I tried to run ESLint on bun but It failed due to this different behavior.

**Comment by jansedlon** [comment]
Just adding that this might be the reason why 'pino' doesn't work in bun. It was throwing errors around working with stack traces as well

**Comment by imZack** [comment]
Was trying to replace node with bun v1.0 but this is the the roadblocker.

**Comment by Voldemat** [comment]
Related: https://github.com/oven-sh/bun/issues/4591

**Comment by mentos1386** [comment]
Related: https://github.com/oven-sh/bun/issues/4742

**Comment by Jarred-Sumner** [comment]
Fixed in https://github.com/oven-sh/bun/pull/5802

This will be released in Bun v1.0.3. 

If you don't want to wait until then, an hour after this comment was written you should be able to upgrade to the canary build of Bun:

```bash
bun upgrade --canary
```
